snuba/datasets/transactions_processor.py

```diff
@@ -37,6 +38,15 @@
 UNKNOWN_SPAN_STATUS = 2
 
 
+@dataclass(frozen=True)
+class StructuredMessage:
```

Comment:
I don't think we need this class. The validate function could return simply the event data and the retention. The other functions can all receive `event` or `event["data"]` as an argument.